compiler/src/main/java/org/qbicc/graph/Node.java

```diff
@@ -752,7 +752,7 @@ public Value visit(final Copier param, final Neg node) {
 
             public Value visit(final Copier param, final New node) {
                 param.copyNode(node.getDependency());
-                return param.getBlockBuilder().new_(node.getClassObjectType());
+                return param.getBlockBuilder().new_(node.getClassObjectType(), node.getTypeId(), node.getSize(), node.getAlign());
```

Comment:
These values should be copied in case they are dynamic e.g. `param.copyValue(node.getTypeId())` etc.

Reply:
Thanks. fixed.